assets/js/script.js

```diff
@@ -0,0 +1,194 @@
+  { timeLabel: "6pm", key: 18 },
+];
+
+//UTILITY FUNCTIONS
+
+//Get from local storage
+const getFromLS = (key) => {
+  return JSON.parse(localStorage.getItem(key));
+};
+
+//Write to local storage
+const writeToLS = (key, data) => {
+  if (data != "") {
+    localStorage.setItem(key, JSON.stringify(data));
+  }
+};
+
+//Remove item from local storage
+const removeFromLS = (key) => {
+  localStorage.removeItem(key);
+};
+
+//Clear local storage
+const clearLS = () => {
+  localStorage.clear();
+};
+
+//END UTILITY FUNCTIONS
+
+//function to render current date in header (moment.js)
+const renderDate = () => {
+  const currentDate = moment().format("dddd, Do of MMMM YYYY");
+  $("#currentDay").append(currentDate);
+};
+
+//function to handle click on remove buttons
+const handleRemoveClick = (event) => {
+  //gets the data-key for the button clicked
+  const targetKey = $(event.target).attr("data-key");
+  //removes the data-key and its value from local storage
+  removeFromLS(targetKey);
+  //re-renders the scheduler
+  renderScheduler();
+};
+
+//function to handle click on save buttons
+const handleSaveClick = (event) => {
+  //gets the data-key for the button clicked
+  const targetKey = $(event.target).attr("data-key");
+  //gets the value of the corresponding textarea (textarea with same data-key)
+  const taskValue = $(`textarea[data-key=${targetKey}]`).val();
+  //sets the data-key and its value in local storage (for data persistence)
+  writeToLS(targetKey, taskValue);
+};
+
+//function to handle click on the clear button
+const handleClearClick = () => {
+  //calls function to clear local storage
+  clearLS();
+  //re-renders the scheduler
+  renderScheduler();
+};
+
+//function to handle a click in the container - redirects to the right handling function
+const handleClick = (event) => {
+  //stops propagation to other elements in the container
+  event.stopPropagation();
+  //gets the date purpose to know if a save or remove button was clicked
+  const targetPurpose = $(event.target).attr("data-purpose");
+  //if remove button was clicked, then goes to the handleRemoveClick function
+  if (targetPurpose == "remove") {
+    handleRemoveClick(event);
+    //if save button was clicked then, goes to the handleSaveClick function
+  } else if (targetPurpose == "save") {
+    handleSaveClick(event);
+  } else if (targetPurpose == "clear") {
+    handleClearClick();
+  }
+};
+
+//function to render the "clear scheduler" button on the page
+const renderClearButton = () => {
+  //creates and appends the button to the container
+  $("#container").append(
+    $("<div>")
+      .addClass("clear-block d-flex flex-row justify-content-center")
+      .append(
+        $("<button>")
+          .addClass("clearBtn p-2")
+          .attr("type", "button")
+          .attr("data-purpose", "clear")
+          .html("Clear Scheduler")
+      )
+  );
+};
+
+//function to render the time blocks on the page
+const renderTimeBlock = () => {
```

Comment:
```suggestion
const renderTimeBlocks = () => {
```
Rename as this function renders multiple time blocks and not just one time block